beep/tests/test_features.py

```diff
@@ -171,20 +173,65 @@ def test_insufficient_data_file(self):
             self.assertEqual(output_obj['message_list'][0]['comment'],
                              'Insufficient or incorrect data for featurization')
 
-    def test_DiagnosticCyclesFeatures_class(self):
+    def test_RPTdQdVFeatures_class(self):
         with ScratchDir('.'):
             os.environ['BEEP_PROCESSING_DIR'] = TEST_FILE_DIR
             pcycler_run_loc = os.path.join(TEST_FILE_DIR, 'PreDiag_000240_000227_truncated_structure.json')
             pcycler_run = loadfn(pcycler_run_loc)
-            featurizer = DiagnosticCyclesFeatures.from_run(pcycler_run_loc, os.getcwd(), pcycler_run)
+            params_dict = {'diag_ref': 0,
+                           'diag_nr': 2,
+                           'charge_y_n': 1,
+                           'rpt_type': 'rpt_2C',
+                           'plotting_y_n': 0}
+            featurizer = RPTdQdVFeatures.from_run(pcycler_run_loc, os.getcwd(), pcycler_run, params_dict)
             path, local_filename = os.path.split(featurizer.name)
             folder = os.path.split(path)[-1]
             dumpfn(featurizer, featurizer.name)
-            self.assertEqual(folder, 'DiagnosticCyclesFeatures')
-            self.assertEqual(featurizer.X.shape[1], 148)
-            self.assertTrue(all(x in featurizer.X.columns for x in ['m0_Mu','variance of ocv','var_charging_dQdV'])
+            self.assertEqual(folder, 'RPTdQdVFeatures')
+            self.assertEqual(featurizer.X.shape[1], 12)
+            self.assertEqual(featurizer.metadata['parameters'], params_dict)
+
+    def test_HPPCResistanceVoltageFeatures_class(self):
+        with ScratchDir('.'):
+            os.environ['BEEP_PROCESSING_DIR'] = TEST_FILE_DIR
+            pcycler_run_loc = os.path.join(TEST_FILE_DIR, 'PreDiag_000240_000227_truncated_structure.json')
+            pcycler_run = loadfn(pcycler_run_loc)
+            featurizer = HPPCResistanceVoltageFeatures.from_run(pcycler_run_loc, os.getcwd(), pcycler_run)
+            path, local_filename = os.path.split(featurizer.name)
+            folder = os.path.split(path)[-1]
+            dumpfn(featurizer, featurizer.name)
+            self.assertEqual(folder, 'HPPCResistanceVoltageFeatures')
+            self.assertEqual(featurizer.X.shape[1], 64)
+            self.assertListEqual([featurizer.X.columns[0], featurizer.X.columns[-1]], ['ohmic_r_d0','D_8'])
+
+
+    def test_HPPCRelaxationFeatures_class(self):
+        with ScratchDir('.'):
+            os.environ['BEEP_PROCESSING_DIR'] = TEST_FILE_DIR
+            pcycler_run_loc = os.path.join(TEST_FILE_DIR, 'PreDiag_000240_000227_truncated_structure.json')
+            pcycler_run = loadfn(pcycler_run_loc)
+            featurizer = HPPCRelaxationFeatures.from_run(pcycler_run_loc, os.getcwd(), pcycler_run)
+            path, local_filename = os.path.split(featurizer.name)
+            folder = os.path.split(path)[-1]
+            dumpfn(featurizer, featurizer.name)
+            self.assertEqual(folder, 'HPPCRelaxationFeatures')
```

Comment:
Would it be possible to include an assert here to check that the meta data contains the correct parameter values?